src/components/pages/Offers/Offers.jsx

```diff
@@ -222,8 +226,46 @@ class Offers extends PureComponent {
     newSelectedOfferIds.length ? showActionsBar() : hideActionsBar()
   }
 
+  selectAllOffers = () => {
+    const {
+      offers,
+      showActionsBar,
+      setSelectedOfferIds,
+      hideActionsBar,
+      selectedOfferIds,
+    } = this.props
+    const { areAllOffersSelected } = this.state
+
+    this.setState({ areAllOffersSelected: !areAllOffersSelected })
+    let newSelectedOfferIds = [...selectedOfferIds]
+
+    if (!areAllOffersSelected) {
+      offers.forEach(offer => newSelectedOfferIds.push(offer.id))
```

Comment:
c'est etrange d'avoir `!areAllOffersSelected` alors je selectionne tout.
```suggestion
    selectAllOffers = () => {
      toggleSelectAllCheckboxes()
      // ...
      const { areAllOffersSelected } = this.state
      const selectedOfferIds = areAllOffersSelected ? offers.map((offer) => offer.id) : []
      selectedOfferIds.length ? hideActionsBar() : showActionsBar()
      setSelectedOfferIds(selectedOfferIds)
    }
```

Reply:
Petit souci d'asynchronisme avec ce refacto (le setState n'était pas fini que la suite était déjà exécutée donc faux négatif) -> ajout d'async await a la fonction selectAllOffers (await du toggleSelectAllCheckboxes())